### PR TITLE
[Refactor] Modify `tools/train.py` and some known bugs

### PR DESCRIPTION
**Pull request by liqikai9**
## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/_base_/default_runtime.py [hunk 1]
```diff
@@ -5,7 +5,9 @@
     timer=dict(type='IterTimerHook'),
     logger=dict(type='LoggerHook', interval=50),
     param_scheduler=dict(type='ParamSchedulerHook'),
-    checkpoint=dict(type='CheckpointHook', interval=1),
+    checkpoint=dict(
+        type='CheckpointHook', interval=1, save_best='coco/AP',
+        rule='greater'),
```
Comment (ly015) [inline configs/_base_/default_runtime.py:10]: `default_runtime.py` will be used by all experiments. It's better not to add coco settings here.

Comment (liqikai9) [reply configs/_base_/default_runtime.py:10]: Agreed, I will move it to coco configs.

File: mmpose/models/heads/heatmap_heads/heatmap_head.py [hunk 2]
```diff
@@ -333,7 +333,12 @@ def loss(self,
             target=to_numpy(gt_heatmaps),
             mask=to_numpy(keypoint_weights) > 0)
 
-        losses.update(acc_pose=float(avg_acc))
+        if torch.cuda.is_available():
```
Comment (ly015) [inline mmpose/models/heads/heatmap_heads/heatmap_head.py:336]: What about getting device from `losses` or `self.parameters()`?

Comment (liqikai9) [reply mmpose/models/heads/heatmap_heads/heatmap_head.py:336]: How about using `device=gt_heatmaps.device`?

**Review by ly015 — APPROVED** [review]

**Pull request merged**